fhir_river_api/src/app.py

```diff
@@ -14,30 +13,47 @@
 
 # Create flask app object
 app = Flask(__name__)
-api = Api(app)
 
 producer = RiverApiProducer(broker=os.getenv("KAFKA_BOOTSTRAP_SERVERS"))
+EXTRACTOR_URL = os.getenv("EXTRACTOR_URL")
+TRANSFORMER_URL = os.getenv("TRANSFORMER_URL")
 
 
-@app.route("/run_sample", methods=["POST"])
+@app.route("/preview", methods=["POST"])
 def trigger_sample_extractor():
     """
     Extract record for the specified resource and primary key value
     """
     body = request.get_json()
     resource_id = body.get("resource_id", None)
     primary_key_values = body.get("primary_key_values", None)
-    batch_id = uuid.uuid4().hex
+    logger.debug("PREVIEW %s %s", resource_id, primary_key_values)
 
     try:
-        create_extractor_trigger(resource_id, batch_id, primary_key_values)
-        return "Success", 200
+        extract_resp = requests.post(
+            f"{EXTRACTOR_URL}/extract",
+            json={"resource_id": resource_id, "primary_key_values": primary_key_values},
+        )
+        if extract_resp.status_code != 200:
+            raise Exception(
+                f"Failed while extracting data: {extract_resp.content.decode('utf-8')}."
+            )
+
+        rows = extract_resp.json()["rows"]
+        transform_resp = requests.post(
+            f"{TRANSFORMER_URL}/transform", json={"resource_id": resource_id, "dataframe": rows[0]},
```

Comment:
Why only rows[0] ? 
Question: could we have many primary_key_values in the preview ?

Reply:
because the preview used to return only one row. However I think it's best to be able to return multiple fhir instances so i'll update that to use lists instead.